react/src/views/Transfers/ShipmentView/ShipmentView.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  Shipment,
+  ShipmentByIdQuery,
+  ShipmentByIdQueryVariables,
+  ShipmentDetail,
+} from "types/generated/graphql";
+import ShipmentCard from "./components/ShipmentCard";
+import ShipmentTabs from "./components/ShipmentTabs";
+
+export const SHIPMENT_BY_ID = gql`
+  query ShipmentById($id: ID! = 1) {
+    shipment(id: $id) {
+      transferAgreement {
+        id
+        type
+      }
+      id
+      state
+      sentBy {
+        id
+        name
+      }
+      sourceBase {
+        id
+        name
+        organisation {
+          id
+          name
+        }
+      }
+      targetBase {
+        id
+        name
+        organisation {
+          id
+          name
+        }
+      }
+      details {
+        box {
+          id
+          labelIdentifier
+          numberOfItems
+          product {
+            id
+            name
+            gender
+            category {
+              id
+              name
+              hasGender
+              sizeRanges {
+                sizes {
+                  id
+                  label
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+function ShipmentView() {
+  // Basics
+  // const navigate = useNavigate();
+  // const { triggerError } = useErrorHandling();
+  // const { createToast } = useNotification();
+  // const { globalPreferences } = useContext(GlobalPreferencesContext);
+
+  // variables in URL
+  const id = useParams<{ id: string }>().id!;
+
+  // Query Data for the Form
+  const shipmentData = useQuery<ShipmentByIdQuery, ShipmentByIdQueryVariables>(SHIPMENT_BY_ID, {
+    variables: {
+      id,
+    },
+  });
+
+  // Handle Loading State
+  if (shipmentData.loading) {
+    return <APILoadingIndicator />;
+  }
+
+  if (shipmentData?.data?.shipment === undefined) {
+    return (
+      <Alert status="error">
+        <AlertIcon />
+        Could not fetch Shipment data! Please try reloading the page.
+      </Alert>
+    );
+  }
+
+  return (
+    <Flex direction="column">
+      <Center>
+        <VStack>
+          <Heading>View Shipment</Heading>
+          <ShipmentCard shipment={shipmentData.data?.shipment as unknown as Shipment} />
+        </VStack>
+      </Center>
+      <Spacer />
+      <Box>
+        <ShipmentTabs
+          shipmentDetail={shipmentData.data?.shipment?.details as unknown as ShipmentDetail[]}
+        />
+      </Box>
+
+      <ButtonGroup gap="4">
+        <Button
+          mt={10}
+          size="md"
+          type="button"
+          borderRadius="0"
+          border={1}
+          borderColor="blackAlpha.800"
+          w="full"
+          variant="solid"
+          backgroundColor="white"
+        >
+          Back to Overview
+        </Button>
+
+        <Button
+          mt={10}
+          type="button"
+          borderRadius="0"
+          w="full"
+          variant="solid"
+          backgroundColor="red.300"
+          color="white"
+        >
+          Reject
```

Comment:
I suppose clicking this button will call `cancelShipment`. Will it be part of another PR?

Reply:
@pylipp need to check this with @aerinsol and @HaGuesto